### ISSUE TITLE
Handle UserEvent in vb.net

### ISSUE THREAD

**Issue by JanWie2** [AsterNET/AsterNET#96]
Dear all,

I have a userevent registered with asterisk and can see it beeing provdid to the client in wireshark dump. But VB will not handle this event but ignores it.
What did I miss?

My class looks like this:
    Public Class ExtCallEvent
    Inherits UserEvent
    Private callername As String = ""
    Private CallerNum As String = ""

    Public Event ExtCallEvent(sender As Object, e As ExtCallEvent)
   
    Property callname() As String
        Get
            Return callername
        End Get
        Set(ByVal Value As String)

            callername = Value

        End Set
    End Property
    Property callnum() As String
        Get
            Return CallerNum
        End Get
        Set(ByVal Value As String)

            CallerNum = Value

        End Set

    End Property
    Sub RaiseEvents()
        RaiseEvent ExtCallEvent(Nothing, Nothing)

    End Sub

    Sub New(source As ManagerConnection)
        MyBase.New(source)
    End Sub
End Class

In the form class I have declared:

WithEvents Manager As New AsterNET.Manager.ManagerConnection

Public Event ExtCallEvent(sender As Object, e As ExtCallEvent)

 Dim WithEvents ueclass As New ExtCallEvent(Manager)

On load I perform:

Manager = New AsterNET.Manager.ManagerConnection(address, port, user, password)
allevents(True)
Manager.Login()
Manager.RegisterUserEventClass(GetType(ExtCallEvent))

My Functions look like:
Public Sub Userevents(sender As Object, e As ExtCallEvent) Handles Manager.UserEvents
msgbox("hi")
End Sub

Public Sub ue_Extcall(sender As Object, e As ExtCallEvent) Handles ueclass.ExtCallEvent
msgbox("hi")
End Sub

But they are not fired when I see the the userdefined event reiggered by
exten => s,n,UserEvent(ExtCallEvent,CallName: ${CALLERID(name)},callNum: ${CALLERID(num)})
and the data in the tcp stream.

Thanks for any help in advance
Jan

**Comment by garysprice** [comment]
seems over complicated to me.
try...

public WithEvents Manager As  AsterNET.Manager.ManagerConnection

     onload

Manager = New AsterNET.Manager.ManagerConnection(address, port, user, password)
allevents(True)
Manager.Login()

Public Sub ue_Extcall(sender As Object, e As AsterNET.Manager.event.ExtCallEvent) Handles Manager.ExtCallEvent
msgbox("hi")
End Sub

(going from memory so forgive mi if you have to tidy some things up...)

**Comment by JanWie2** [comment]
Thanks! 
The main difference lies in 
`Handles Manager.ExtCallEvent` instead of using `ueclass.ExtCallEvent`.
I think this might be the solution and I tried it before but it does not work as `ExtCallEvent` is not an event of `Manager`.
I have registered the event with asterisk using manger. But this seems not to make it part of manager. How can I solve this?

But even without such  handle I would have expected to find the `ExtCallEvent` with 
`Private Sub unhandledEvent(sender As Object, e As AsterNET.Manager.Event.ManagerEvent) Handles Manager.UnhandledEvent`  or  `Private Sub unhandledUserEvent(sender As Object, e As ExtCallEvent) Handles Manager.UnhandledEvent`. But I do not.
Moreover using `Private Sub unhandledUserEvent(sender As Object, e As ExtCallEvent) Handles Manager.UnhandledEvent` or `Public Sub userevents(sender As Object, e As Object) Handles Manager.UserEvents` leeds to a disconnect from AMI.

But I found a **solution/workaround**: 
Using `Private Sub unhandledEvent(sender As Object, e As Object) Handles Manager.UnhandledEvent`
handles the  `ExtCallEvent` user event as well an I can work from there, filtering it and handling it.
